src/celeritas/optical/surface/SimpleReflectionExecutor.hh

```diff
@@ -0,0 +1,93 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/SimpleReflectionExecutor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "celeritas/geo/GeoFwd.hh"
+#include "celeritas/optical/CoreTrackView.hh"
+
+#include "SurfaceInteraction.hh"
+#include "SurfacePhysicsData.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ */
+struct SimpleReflectionExecutor
+{
+    inline CELER_FUNCTION SurfaceInteraction operator()(CoreTrackView&) const;
+
+    NativeCRef<SurfacePhysicsParamsData> params;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ */
+CELER_FUNCTION SurfaceInteraction
+SimpleReflectionExecutor::operator()(CoreTrackView& track) const
+{
+    // TODO: add assertions
+
+    auto geo = track.geometry();
+    auto const& direction = geo.dir();
+    auto const& polarization = track.particle().polarization();
+    auto rng = track.rng();
+
+    // Get surface normal from geometry
+    Real3 surface_normal = geo.global_surface_normal();
+
+    real_type reflectivity = params.scalars.global_reflectivity;
+    real_type refl_and_trans = reflectivity
+                               + params.scalars.global_transmittance;
+
+    CELER_EXPECT(0 <= reflectivity && reflectivity <= refl_and_trans
```

Comment:
Easier to read if `reflectivity >= 0`.

Reply:
Yeah, I don't really object to as well. My preference comes from my brain always defaulting to [my_parameter] on the left and [what_should_be] on the right when reading.